DS-2986 Handle Aliased Dummy variables

jrwishart opened the pull request:
When Dummy Variable Adjustment is used in Regression, it needs to check for 
predictors that have identical structure in their missing values. This 
PR handles the deduplicated dummy variables handled by flipData and warns the 
user if any dummy variables are aliased. It also correctly remaps the data suitable
for RIA/Shapley.

- DS-2986 Handle aliased dummy variables
- DS-2986 Add unit tests for aliased dummy vars
- DS-2986 Point to flipData@DS-2986 in .travis.yml

jrwishart committed:
DS-2986 Handle aliased dummy variables

Added helper function to throw warnings for aliased dummmy variables and
also updated the remapping/adjustment of the original data based off the
dummy variables to handle the case of aliased dummy variables using the
new attribute containing the variable mapping in flipData.

jrwishart committed:
DS-2986 Add unit tests for aliased dummy vars

jrwishart committed:
DS-2986 Point to flipData@DS-2986 in .travis.yml

jrwishart committed:
DS-2986 Version adjustment in DESCRIPTION

jrwishart committed:
DS-2896 Update DESC to point at flipData@DS-2986

Need to point at the development branch of flipData in the downstream
builds of flipRegression in flipCluster, flipAnalaysisofVariance etc.

jrwishart committed:
fixup! DS-2896 Update DESC to point at flipData@DS-2986

mwmclean commented:
Seems to be an error unfortunately using this in Displayr with multiple data sets containing the same variable name as one used in the regression (e.g. necessitating use `dat$Questions$x` in formulas instead of just `x`)  https://app.displayr.com/Dashboard?project_id=-240605#page=13af8857-46fa-4ffb-9636-5b1f5b65243b The other missing data methods don't error.

jrwishart commented:
> Seems to be an error unfortunately using this in Displayr with multiple data sets containing the same variable name as one used in the regression (e.g. necessitating use `dat$Questions$x` in formulas instead of just `x`) https://app.displayr.com/Dashboard?project_id=-240605#page=13af8857-46fa-4ffb-9636-5b1f5b65243b The other missing data methods don't error.

Fixed the issues that caused this. I updated the `flipData` functions to handle dummy variable adjustment. The updates are now in https://github.com/Displayr/flipData/pull/8

jrwishart committed:
DS-2986 Fix typo and improve tests in aliased dummy

Fixed typo in the warning thrown for aliased dummy variables. Also
improved the tests to cover a three group scenario and also tested the
correct display of variable labels when requested.

Commit:
Fix typo in warning msg; DS-2986 [revdep skip]

Commit:
Rm need for flipData branch; DS-2986 [ci skip]
